tango/step_cache.py

```diff
@@ -48,6 +48,11 @@ def __setitem__(self, step: Step, value: Any) -> None:
         """Writes the results for the given step. Throws an exception if the step is already cached."""
         raise NotImplementedError()
 
+    @abstractmethod
+    def __delitem__(self, step_unique_id) -> None:
```

Comment:
We need type annotation here (and at other places that implement this function).

Also, let's have consistency in setitem and delitem. If setitem takes Step object as input, so should delitem.